scripts/flow_track_correl.py

```diff
@@ -0,0 +1,414 @@
+    """
+    Read the header from a FITS file.
+
+    Parameters
+    ----------
+    filepath : str
+        Path to the FITS file.
+    hdu : int, optional
+        Index of the header data unit containing the header information, by default 1.
+
+    Returns
+    -------
+    fits.header.Header
+        Header object from the specified FITS file.
+    """
+    with fits.open(filepath) as hdul:
+        hdul.verify('fix')
+        header = hdul[hdu].header
+    return header
+
+
+def calc_ylims(ycen_band_rs: np.ndarray, band_width: float, header: fits.header.Header) -> [int, int]:
+    """
+    Convert y-coordinates of lower and upper row of bands to array indices for slicing.
+
+    Parameters
+    ----------
+    ycen_band_rs : np.ndarray
+        y-coordinates of center of band in solar radii.
+
+    band_width : float
+        Half-width of the band in solar radii.
+
+    header : fits.header.Header
+        FITS header containing image information.
+
+    Returns
+    -------
+    list
+        Lower and upper Numpy array indices of the band.
+    """
+    y0_image_px = 1
+    y0_cen_px = header['CRPIX2']
+    y0_cen_rs = header['CRVAL2']
+    px_scale = header['CDELT2']
+    origin_rs = y0_cen_rs + (y0_image_px - y0_cen_px) * px_scale
+    ylo_band_idx = ((ycen_band_rs - band_width) - origin_rs) / px_scale
+    yhi_band_idx = ((ycen_band_rs + band_width) - origin_rs) / px_scale
+    return [ylo_band_idx, yhi_band_idx]
+
+
+def preprocess_image(image: np.ndarray) -> np.ndarray:
+    """
+    Normalize and preprocess FITS image by removing bad values and scaling.
+
+    Parameters
+    ----------
+    image : np.ndarray
+        Input FITS image array.
+
+    Returns
+    -------
+    np.ndarray
+        Preprocessed image array.
+    """
+    image[np.abs(image) > 100] = 0
+    image[~np.isfinite(image)] = 0
+    image -= np.mean(image, axis=0)
+    return image / np.sqrt(np.mean(image ** 2, axis=0))
+
+
+def calculate_cross_correlation(a: np.ndarray, b: np.ndarray, offsets: np.ndarray, dr: int, of0: int) -> np.ndarray:
+    """
+    Perform cross-correlation for a range of offsets.
+
+    Parameters
+    ----------
+    a : np.ndarray
+        First image array for correlation.
+
+    b : np.ndarray
+        Second, time-offseted image array for correlation.
+
+    offsets : np.ndarray
+        Array of pixel offsets to iterate over for cross-correlation.
+
+    dr : int
+        Pixel offset increment between samples.
+
+    of0 : int
+        Central offset to start correlation from.
+
+    Returns
+    -------
+    np.ndarray
+        Accumulated cross-correlation array over all offsets.
+    """
+    acc = np.zeros((len(offsets), a.shape[0], a.shape[1]), dtype=float)
+    for jj, offset_index in enumerate(offsets):
+        this_of = int(dr * (offset_index - (len(offsets) - 1) / 2)) + of0
+        ofa = int(this_of / 2)
+        ofb = int(this_of) - ofa
+
+        if ofa < 0:
```

Comment:
Could you add a comment in the code on where these pad shapes come from? It'll be helpful for when we return to the code later.

Reply:
Will do. That was a rather "arbitrary" decision from Craig's original code to choose in which direction to shift the images with respect to each other (the first value is... 42... lol), but depending on the dimensions of the polar-remapped data of the mosaic data, I may adapt that for both clarity and efficiency.

Reply:
42. The answer!

Reply:
Will add comment on the padding too, not just the values of ofa and ofb